File: prolog/euclid.pl
evaluated_gcd(A, B, Div) :-
  A < B -> gcd(B, A, Div)
  ; 0 is B -> Div is A
  ; Diff is A mod B, evaluated_gcd(Diff, B, Div)
.

gcd(A, B, Div) :-
  EvalA is A, EvalB is B, evaluated_gcd(EvalA, EvalB, Div)
.

lcm(A, B, LCM) :-
  gcd(A, B, Div),
  LCM is A * B / Div
.

File: project_euler/rust/src/euclid.rs
pub fn gcd(a:u64, b:u64) -> u64 {
  if b < a {
    return gcd(b, a);
  }
  let remainder = b % a;
  if remainder == 0 {
    return a;
  }
  return gcd(remainder, a);
}


pub fn lcm(a:u64, b:u64) -> u64 {
  return (a * b) / gcd(a, b);
}
